google-sheets/trialSheet.py

```diff
@@ -0,0 +1,55 @@
+import os
+# imports
+from google.auth.transport.requests import Request
+from google.oauth2.credentials import Credentials
+from google_auth_oauthlib.flow import InstalledAppFlow
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+
+class GoogleSheetsAPI:
+    def __init__(self, credentials="token1.json", key ="./credentials.json" , SPREAD_ID = None):
+        self.SCOPES = ["https://www.googleapis.com/auth/spreadsheets"]  #accessed by multiple functions in multiple places 
+        self.credentials = credentials
+        self.key = key
+        self.SPREAD_ID = SPREAD_ID # we leave this parameter as none since whenever a new object is specified this will be unique identifier of it
+
+    def auth(self):
```

Comment:
this can be done in the Init itself try to implement there only you can take the reference from the other function file

Reply:
I assume in this you are talking about the different libraries I have imported. I tried importing the libaries inside the init function but then whatever these libraries have are not readable by other functions anymore. So for now I have left this code untouched.